### PR TITLE
feat: add norm

### PR DESCRIPTION
**Pull request by Soptq**
This PR adds `norm` operator.

## Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?

- calculating Norm of a given list.
- all tests are passed.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/linalg/src/norm.cairo [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+use alexandria_math::pow;
+
+/// Compute the norm for an u128 array.
+/// # Arguments
+/// * `array` - The inputted array.
+/// * `ord` - The order of the norm.
+/// * `iter` - The number of iterations to run the algorithm
+/// # Returns
+/// * `u128` - The norm for the array.
+fn norm(mut xs: Span<u128>, ord: u128, iter: usize) -> u128 {
```
Comment (LucasLvy) [inline src/linalg/src/norm.cairo:12]: `Into<T, u128`?

**Review by LucasLvy — CHANGES_REQUESTED** [review]

File: src/linalg/src/norm.cairo (hunk 1 above)
Comment (Soptq) [reply src/linalg/src/norm.cairo]: Fixed ;)

File: src/linalg/src/norm.cairo [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        };
+    };
+
+    if ord == 0 {
+        return norm;
+    }
+
+    norm = fast_nr_optimize(norm, ord, iter);
+
+    norm
```
Comment (LucasLvy) [inline src/linalg/src/norm.cairo:37]:
```suggestion
    fast_nr_optimize(norm, ord, iter)
```

**Review by LucasLvy — CHANGES_REQUESTED** [review]

**Review by LucasLvy — APPROVED** [review]

File: src/linalg/src/tests/norm_test.cairo [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+impl u128_to_u32 of Into<u128, u32> {
+    fn into(self: u128) -> u32 {
+        self.try_into().unwrap()
+    }
+}
+
+#[test]
+#[available_gas(2000000)]
+fn norm_test_1() {
+    let mut array: Array<u128> = array![3, 4];
```
[the remark this answers is not in the document]
Comment (Soptq) [reply src/linalg/src/tests/norm_test.cairo:6]: Thanks for reviewing, this is fixed in this PR: https://github.com/keep-starknet-strange/alexandria/pull/268

**Pull request merged**